backend/app/search/connect.py

```diff
@@ -15,10 +17,13 @@ def connect_elasticsearch():
     _es = None
     logger.info(settings.elasticsearch_url)
     _es = Elasticsearch(settings.elasticsearch_url)
-    if _es.ping():
-        logger.info("Successfully connected to Elasticsearch")
-    else:
-        logger.info("Can not connect to Elasticsearch")
+    try:
+        if _es.ping():
+            logger.info("Successfully connected to Elasticsearch")
+        else:
+            raise ServiceUnreachable("Elasticsearch")
+    except ServiceUnreachable as e:
+        log_error(e)
```

Comment:
i think you are missing an `await` here. log_error function is async. I set an break point here and it didn't actually enter this function :-(

Reply:
thank you, i added the await. we can talk today about how to test this